sdk/cosmos/azure-cosmos/azure/cosmos/partition_key.py

```diff
@@ -102,58 +102,55 @@ class PartitionKey(dict):
     """
 
     @overload
-    def __init__(self, path: list, *, kind: Literal["MultiHash"] = "MultiHash", version: int = 2) -> None:
-        self.path = path
-        self.kind = kind
-        self.version = version
+    def __init__(self, path: Sequence[str], *, kind: Literal["MultiHash"] = "MultiHash", version: int = 2) -> None:
+        ...
 
     @overload
     def __init__(self, path: str, *, kind: Literal["Hash"] = "Hash", version: int = 2) -> None:
-        self.path = path
-        self.kind = kind
-        self.version = version
+        ...
 
-    def __init__(self, *args, **kwargs):  # pylint: disable=super-init-not-called
-        self.path = args[0] if args else kwargs['path']
-        self.kind = args[1] if len(args) > 1 else kwargs.get('kind', 'Hash')
-        self.version = args[2] if len(args) > 2 else kwargs.get('version', 2)
+    def __init__(self, *args, **kwargs):
+        path = args[0] if args else kwargs['path']
+        kind = args[1] if len(args) > 1 else kwargs.get('kind', 'Hash' if isinstance(path, str) else 'MultiHash')
```

Comment:
This behaviour has changed slightly in that I automatically interpret a `str` to be a `Hash` type and a `Sequence` to be a `multihash` type - this could be a problem if there's intentions in future to add additional `kinds` that will make this distinction difficult.

Reply:
This should be fine, if we were to need another kind, this whole class would require additional modifications to accommodate that.